### PR TITLE
Add support for git push with custom headers

### PR DESCRIPTION
**Pull request by evolvedlight**
This is needed for pushing changes to bitbucket server repos

### CODE REVIEW

**Review by JackLiang5 — APPROVED** [review]

File: LibGit2Sharp/Network.cs [hunk 1]
```diff
@@ -376,6 +376,7 @@ public virtual void Push(Remote remote, IEnumerable<string> pushRefSpecs, PushOp
                                           PackbuilderDegreeOfParallelism = pushOptions.PackbuilderDegreeOfParallelism,
                                           RemoteCallbacks = gitCallbacks,
                                           ProxyOptions = new GitProxyOptions { Version = 1 },
+                                          CustomHeaders = pushOptions.CustomHeaders?.Length > 0 ? GitStrArrayManaged.BuildFrom(pushOptions.CustomHeaders) : new GitStrArrayManaged()
```
Comment (JackLiang5) [inline LibGit2Sharp/Network.cs:379]: Is null better than new GitStrArrayManaged() here？

**Comment by TackyTortoise** [comment]
Without this it is also seemingly impossible to push to Azure repository using a personal access token. I wonder why it was exposed in FetchOptions but not in PushOptions.

File: LibGit2Sharp/Network.cs (hunk 1 above)
Comment (TackyTortoise) [reply LibGit2Sharp/Network.cs:379]: The CustomHeaders field is of type GitStrArrayManaged, which is a struct type so non-nullable value type. So not without some other impacting modifications I think.

Comment (James-LG) [inline LibGit2Sharp/Network.cs:379]: `GitStrArrayManaged` needs to be disposed. I have a similar PR that does that here #2052 by following the pattern set by `FetchOptionsWrapper`.

**Comment by evolvedlight** [comment]
Sorry, somehow I missed the notifications on the comment. Will close in favor of https://github.com/libgit2/libgit2sharp/pull/2052

**Pull request closed without merging**